File: src/main/kotlin/TwoSum.kt
fun twoSum(nums: IntArray, target: Int): IntArray {
    val map = hashMapOf<Int, Int>()
    nums.forEachIndexed { index, value ->
        val want = map[target - value]
        if (want != null) {
            return intArrayOf(want, index)
        }
        map[value] = index
    }
    return IntArray(0)
}

File: golang/TwoSum.go
package main

func twoSum(nums []int, target int) []int {
	m := make(map[int]int, len(nums))
	for i, num := range nums {
		t := m[target-num]
		if t != 0 {
			return []int{i, t - 1}
		} else {
			m[num] = i + 1
		}
	}
	return nil
}
